CONTRIBUTING.md

```diff
@@ -0,0 +1,79 @@
+Contributing to NoteX
+=====================
+
+NoteX is open-source and you're welcome to contribute to it. You're encouraged to submit [pull requests](https://github.com/NoteXCommunity/note-x/pulls), [propose features and discuss issues](https://github.com/NoteXCommunity/note-x/issues).
+
+### Clone the Project
+Clone the project on your local machine using the commands
+```
+git clone https://github.com/NoteXCommunity/note-x.git
+cd note-x
+git remote add upstream https://github.com/NoteXCommunity/note-x.git
+```
+### Create a Branch for your feature
+
+Make sure your branch is up-to-date with ```dev``` and create a branch based on your name and whether it is a bugfix or a feature.  (The name `name/feature/my-feature` or `name/bugfix/foo-button-fixed` is an example). 
+```
+Please use the convention -> your-name/feature/feature-name or your-name/bugfix/bug-name
```

Comment:
nit: @hadiyarajesh as Discussed this was decided to keep as **Github Username**. Because that is referential in the case of **Github Repo**. _(There are several people who doesn't know everyone with the name)_

```suggestion
Please use the convention -> github-username/feature/feature-name or your-name/bugfix/bug-name
```

Reply:
@sumanabhi GitHub username is something we should use, but I've noticed some have some caps-username, some have - in name. That would create inconsistency. To handle this, we may need to ask them to change the GitHub username.